compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageCompilation.java

```diff
@@ -209,8 +210,12 @@ private void compileModulesInternal() {
             for (ModuleContext moduleContext : packageResolution.topologicallySortedModuleList()) {
                 moduleContext.compile(compilerContext);
                 for (Diagnostic diagnostic : moduleContext.diagnostics()) {
-                    diagnostics.add(new PackageDiagnostic(diagnostic, moduleContext.descriptor(),
-                            moduleContext.project()));
+                    if (!ProjectKind.BALA_PROJECT.equals(moduleContext.project().kind()) ||
+                            (diagnostic.diagnosticInfo().severity() == DiagnosticSeverity.ERROR) ||
+                            (diagnostic.diagnosticInfo().severity() == DiagnosticSeverity.INTERNAL)) {
```

Comment:
Why do we need the INTERNAL diagnostics?

Reply:
I mistakenly took it as an error.. I think we can remove it.